src/app/todo/todo-form/todo-form.component.ts

```diff
@@ -1,18 +1,46 @@
 import { Component, OnInit} from '@angular/core';
-import { Todo } from '../models/todo'
-import {FormControl, Validators} from '@angular/forms';
-import {NgbDateStruct} from '@ng-bootstrap/ng-bootstrap';
-import { TodoService } from '../../services/todo.service';
+import { Todo } from '../models/todo';
 import { Output, EventEmitter } from '@angular/core';
+import {NgbModal, ModalDismissReasons} from '@ng-bootstrap/ng-bootstrap';
+import Swal from 'sweetalert2';
 
 @Component({
   selector: 'app-todo-form',
   templateUrl: './todo-form.component.html',
   styleUrls: ['./todo-form.component.css']
 })
 export class TodoFormComponent {
+  constructor(private modalService: NgbModal) {}
+
   @Output() newItemEvent = new EventEmitter<Todo>();
   todo = {title:'', description:'', deadline: new Date()}; 
+  IsmodelShow = false;
```

Comment:
maybe use `deactivate guard`?

Reply:
I used another method to fix this issue and submitted to this branch